platform/darwin/src/MGLAttributionInfo.mm

```diff
@@ -65,10 +65,15 @@ @implementation MGLAttributionInfo
     NSData *htmlData = [styledHTML dataUsingEncoding:NSUTF8StringEncoding];
 
 #if TARGET_OS_IPHONE
-    NSMutableAttributedString *attributedString = [[NSMutableAttributedString alloc] initWithData:htmlData
-                                                                                          options:options
-                                                                               documentAttributes:nil
-                                                                                            error:NULL];
+    __block NSMutableAttributedString *attributedString;
+    
+    dispatch_sync(dispatch_get_main_queue() , ^{
```

Comment:
Note that this method is called as part of `-[MGLTileSource attributionInfos]`, which is a public method that can be called at any time. Even internal uses of this method tend to occur in a for loop, since there are multiple sources to inspect at the same time.

If performance does turn out to be a problem, an alternative to blocking on the main thread here would be to do so further up the call stack, wherever it’s expected that the caller could be on a thread other than the main thread:

https://github.com/mapbox/mapbox-gl-native/blob/bb55bae76db4454affa2cacca956e726a2b43a4b/platform/darwin/src/MGLMapSnapshotter.mm#L213

This would require documenting other callers, such as `-[MGLTileSource attributionInfos]`, as needing to run on the main thread. There is precedent for such a requirement.

Reply:
Updated to check whether it's being called on the main queue. Would it be better to call the `NSMutableAttributedString` initializer earlier than to fiddle with queues? @julianrex and I found that the exception did not occur when we created an `NSMutableAttributedString` in the initializer for a map view.

Reply:
> @julianrex and I found that the exception did not occur when we created an NSMutableAttributedString in the initializer for a map view.

We can't rely on that behavior as it's not documented.

Reply:
Sorry if I'm just parading my iOS ignorance here, but isn't `MGLMapView` essentially required to be created and used on the main thread? And if that's the case then creating the attributed string in the initializer should be good, right?

If we can find a way to avoid "fiddling with queues" we should. It's not the queues themselves that are the problem -- but hidden/unexpected cross-thread blocking calls are trouble.

Reply:
@ChrisLoer @jmkiley was referring to the fact that we can avoid the crash if we initialize any old `NSMutableAttributedString` on the main thread (it ends up calling `WebKitInitialize` using a `dispatch_once` - I suspect it's that initializer that needs to be on the main thread) - not the specific one.